src/specializations/Tetrahedron.jl

```diff
@@ -1,55 +1,36 @@
-################################################################################
+############################################################################################
 #                  Specialized Methods for Tetrahedron
 #
 # Why Specialized?
-#   The Tetrahedron geometry is a volumetric simplex whose parametric function
-#   in Meshes.jl uses barycentric coordinates on a domain {u,v,w} with coordinates
-#   that are non-negative and bound by the surface $u + v + w ≤ 1$. This requires
-#   a multi-step domain transformation whose derivation is detailed in the package
-#   documentation.
-################################################################################
+#   The Tetrahedron geometry is a volumetric simplex whose parametric function in Meshes.jl
+#   uses barycentric coordinates on a domain {u,v,w} with coordinates that are non-negative
+#   and bounded by the surface $u + v + w ≤ 1$. A transformation is used to map this volume
+#   to a rectangular domain [0,1]^3.
+############################################################################################
 
 function integral(
         f::F,
         tetrahedron::Meshes.Tetrahedron,
-        rule::GaussLegendre;
-        diff_method::DM = Analytical(),
-        FP::Type{T} = Float64
-) where {F <: Function, DM <: DifferentiationMethod, T <: AbstractFloat}
-    _error_unsupported_combination("Tetrahedron", "GaussLegendre")
+        rule::IntegrationRule;
+        kwargs...
+) where {F <: Function}
+    paramfunction(t1, t2, t3) = _parametric(tetrahedron, t1, t2, t3)
+    tetra = _ParametricGeometry(paramfunction, 3)
+    return _integral(f, tetra, rule; kwargs...)
 end
 
-function integral(
-        f::F,
-        tetrahedron::Meshes.Tetrahedron,
-        rule::GaussKronrod;
-        diff_method::DM = Analytical(),
-        FP::Type{T} = Float64
-) where {F <: Function, DM <: DifferentiationMethod, T <: AbstractFloat}
-    _guarantee_analytical(Meshes.Tetrahedron, diff_method)
+################################################################################
+#                               Parametric
+################################################################################
 
-    nil = zero(FP)
-    ∫uvw(u, v, w) = f(tetrahedron(u, v, w))
-    ∫vw(v, w) = QuadGK.quadgk(u -> ∫uvw(u, v, w), nil, FP(1 - v - w); rule.kwargs...)[1]
-    ∫w(w) = QuadGK.quadgk(v -> ∫vw(v, w), nil, FP(1 - w); rule.kwargs...)[1]
-    ∫ = QuadGK.quadgk(∫w, nil, one(FP); rule.kwargs...)[1]
+function _parametric(tetrahedron::Meshes.Tetrahedron, t1, t2, t3)
+    @assert (0 <= t3 <= 1)
```

Comment:
I only tested for `t3` here because `t1` and `t2` get checked prior to use in Triangle’s `_parametric` function.

Reply:
I left off the user-friendly error messages in these functions because these geometries aren’t meant to be used beyond the six new hard-coded specializations present in this package. In that sense, I’d actually considered dropping even the `assert`s once I was confident the code was working as intended. However, performance seems pretty good already so I don’t see a strong reason to not just use a friendly error condition.